app/forms/date_form.rb

```diff
@@ -3,8 +3,10 @@
 # Form object for handling date input
 class DateForm < ApplicationForm
   attribute :year, :integer
-  validates :year,
-            numericality: { greater_than_or_equal_to: 1000, less_than_or_equal_to: Time.zone.now.year, allow_nil: true }
+  validates :year, numericality: { greater_than_or_equal_to: 1000, allow_nil: true }
+  validates :year, numericality: { less_than_or_equal_to: Time.zone.now.year,
+                                   allow_nil: true,
+                                   message: 'Must be in the past' } # rubocop:disable Rails/I18nLocaleTexts
```

Comment:
For consistency, "must" should be lowercase. Also for consistency, it should be in en.yml.